pkg/controller/deployment/util/deployment_util.go

```diff
@@ -612,9 +612,10 @@ func EqualIgnoreHash(template1, template2 v1.PodTemplateSpec) bool {
 // FindNewReplicaSet returns the new RS this given deployment targets (the one with the same pod template).
 func FindNewReplicaSet(deployment *extensions.Deployment, rsList []*extensions.ReplicaSet) (*extensions.ReplicaSet, error) {
 	newRSTemplate := GetNewReplicaSetTemplate(deployment)
+	sort.Sort(controller.ReplicaSetsByCreationTimestamp(rsList))
```

Comment:
@smarterclayton @mfojtik we are going to choose the oldest replica set out of two identical replica sets for now - seems like the sanest thing to do when we end up in such a state (seems possible after cluster upgrades: https://github.com/kubernetes/kubernetes/issues/40415).